### ISSUE TITLE
Merge of ea400129c without fef8bd41f into zfs-2.1-release broke i386 build

### ISSUE THREAD

**Issue by mmatuska** [openzfs/zfs#12244]
### System information
FreeBSD stable/13 i386

### Describe the problem you're observing
The changes in ea400129c376c958e32bd912ea29905107ebe0bb (More aggsum optimizations) expect the commit fef8bd41fc178d7212957b611c9bc81fe10cb63e (libspl: implement atomics in terms of atomics) to be landed.

Merge of ea400129c376c958e32bd912ea29905107ebe0bb without fef8bd41fc178d7212957b611c9bc81fe10cb63e broke the build on i386
The code does not build on i386.

Suggested solutions:
Option 1) merge fef8bd41fc178d7212957b611c9bc81fe10cb63e into zfs-2.1-release
Option 2) add patch to asm-i386/atomic.S from Alexander Motin: https://people.freebsd.org/~mav/zzz3S.patch

### Describe how to reproduce the problem
Build OpenZFS 2.1.0-rc7 on i386

### Include any warning/errors/backtraces from the system logs
```
ld: error: undefined symbol: atomic_load_64
>>> referenced by aggsum.c:119 (../../module/zfs/aggsum.c:119)
>>>               .libs/aggsum.o:(aggsum_lower_bound)
>>> referenced by aggsum.c:125 (../../module/zfs/aggsum.c:125)
>>>               .libs/aggsum.o:(aggsum_upper_bound)
>>> referenced by aggsum.c:218 (../../module/zfs/aggsum.c:218)
>>>               .libs/aggsum.o:(aggsum_compare)
>>> referenced 1 more times

ld: error: undefined symbol: atomic_store_64
>>> referenced by aggsum.c:157 (../../module/zfs/aggsum.c:157)
>>>               .libs/aggsum.o:(aggsum_value)
>>> referenced by aggsum.c:158 (../../module/zfs/aggsum.c:158)
>>>               .libs/aggsum.o:(aggsum_value)
>>> referenced by aggsum.c:199 (../../module/zfs/aggsum.c:199)
>>>               .libs/aggsum.o:(aggsum_add)
>>> referenced 3 more times
cc: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by mmatuska** [comment]
@behlendorf please take a look into this

**Comment by behlendorf** [comment]
I've gone ahead and applied the alternate fix of applying all of fef8bd4 to the 2.1 release branch.
